### PR TITLE
switch to new elasticsearch-js for users things

### PR DESCRIPTION
**Pull request by awick**
**Clearly describe the problem and solution**

**Relevant issue number(s) if applicable**

**Did you run `npm run lint` from the viewer or parliament directory (whichever you are making changes to) and correct any errors**

**Did you Ensure that all tests still pass by navigating to the `tests` directory and running `./tests.pl --viewer`**

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: viewer/db.js [hunk 1]
```diff
@@ -735,14 +732,23 @@ exports.flushCache = function () {
...
 
+// Return a user from DB, callback only
 exports.getUser = function (name, cb) {
-  return internals.usersElasticSearchClient.get({ index: internals.usersPrefix + 'users', id: name }, cb);
+  internals.usersClient7.get({ index: internals.usersPrefix + 'users', id: name }, (err, result) => {
+    cb(err, result.body || { found: false });
+  });
 };
 
+// Return a user from cache, callback only
```
Comment (31453) [inline viewer/db.js:751]: Why not use a promise here too?

File: viewer/db.js [hunk 2]
```diff
@@ -760,21 +766,28 @@ exports.getUserCache = function (name, cb) {
...
+// Return a user from cache, promise only
+exports.numberOfUsers = function () {
+  return new Promise((resolve, reject) => {
+    internals.usersClient7.count({ index: internals.usersPrefix + 'users', ignoreUnavailable: true })
+      .then((results) => { resolve({ count: results.body.count }); })
+      .catch((error) => { reject(error); });
+  });
 };
 
+// Delete user, callback only
```
Comment (31453) [inline viewer/db.js:778]: Why not use a promise here too?

File: viewer/db.js [hunk 3]
```diff
@@ -760,21 +766,28 @@ exports.getUserCache = function (name, cb) {
...
 exports.deleteUser = function (name, cb) {
   delete internals.usersCache[name];
-  return internals.usersElasticSearchClient.delete({ index: internals.usersPrefix + 'users', id: name, refresh: true }, (err) => {
+  internals.usersClient7.delete({ index: internals.usersPrefix + 'users', id: name, refresh: true }, (err) => {
     delete internals.usersCache[name]; // Delete again after db says its done refreshing
     cb(err);
   });
 };
 
+// Set user, callback only
```
Comment (31453) [inline viewer/db.js:787]: Why not use a promise here too?

**Review by 31453 — APPROVED** [review]
Works well for me. Would be better if everything used promises though 😃

**Comment by awick** [comment]
maybe when those apis are being moved? :) i just wanted to convert something and see about gotchas :)

**Pull request merged**